aspnetcore/host-and-deploy/windows-service.md

```diff
@@ -5,56 +5,83 @@ description: Learn how to host an ASP.NET Core app in a Windows Service.
-> ```
-
 ## Deployment type
 
-You can create either a framework-dependent or self-contained Windows Service deployment. For information and advice on deployment scenarios, see [.NET Core application deployment](/dotnet/core/deploying/).
+Create either a framework-dependent or self-contained Windows Service deployment. For information and advice on deployment scenarios, see [.NET Core application deployment](/dotnet/core/deploying/).
 
-### Framework-dependent deployment
+### Framework-dependent deployment (FDD)
 
-Framework-dependent deployment (FDD) relies on the presence of a shared system-wide version of .NET Core on the target system. When the FDD scenario is used with an ASP.NET Core Windows Service app, the SDK produces an executable (*\*.exe*), called a *framework-dependent executable*.
+Framework-dependent deployment (FDD) relies on the presence of a shared system-wide version of .NET Core on the target system. When the FDD scenario is adopted following the guidance in this article, the SDK produces an executable (*.exe*), called a *framework-dependent executable*.
 
-### Self-contained deployment
+### Self-contained deployment (SCD)
 
-Self-contained deployment (SCD) doesn't rely on the presence of shared components on the target system. The runtime and the app's dependencies are deployed with the app to the hosting system.
+Self-contained deployment (SCD) doesn't rely on the presence of a shared framework on the host system. The runtime and the app's dependencies are deployed with the app.
 
 ## Convert a project into a Windows Service
 
 Make the following changes to an existing ASP.NET Core project to run the app as a service:
 
 ### Project file updates
 
-Based on your choice of [deployment type](#deployment-type), update the project file:
+Based on your choice of [deployment type](#deployment-type), update the project file according to the following guidance.
+
+#### Package references
 
-#### Framework-dependent Deployment (FDD)
+::: moniker range=">= aspnetcore-3.0"
 
-Add a Windows [Runtime Identifier (RID)](/dotnet/core/rid-catalog) to the `<PropertyGroup>` that contains the target framework. In the following example, the RID is set to `win7-x64`. Add the `<SelfContained>` property set to `false`. These properties instruct the SDK to generate an executable (*.exe*) file for Windows.
+Add a package reference for [Microsoft.Extensions.Hosting.WindowsServices](https://www.nuget.org/packages/Microsoft.Extensions.Hosting.WindowsServices).
+
+::: moniker-end
+
+::: moniker range="< aspnetcore-3.0"
+
+Add a package reference for [Microsoft.AspNetCore.Hosting.WindowsServices](https://www.nuget.org/packages/Microsoft.AspNetCore.Hosting.WindowsServices). To enable Windows Event Log logging, add a package reference for [Microsoft.Extensions.Logging.EventLog](https://www.nuget.org/packages/Microsoft.Extensions.Logging.EventLog).
+
+::: moniker-end
+
+#### Framework-dependent Deployment
```

Comment:
Problem in the past was that devs said that they needed the explicit instructions for either FDD or SCD. It wasn't possible to allow the normal docs to cover it because special MSBuild properties were (and still are for 2.1 and 2.2) required. I can cut it all out, and we can just go with FDD (FDE to be exact). I guess I would just tell anyone asking for 2.1/2.2 SCD help to refer to this last commit on this PR. Shall I proceed with the 🔪 ... go FDE only?